ui/src/common/components/CarouselModal/CarouselModal.jsx

```diff
@@ -0,0 +1,151 @@
+import React, { useCallback, useState, useMemo, forwardRef } from 'react';
+import PropTypes from 'prop-types';
+import Modal from 'react-modal';
+import { Carousel, Row, Col, Button } from 'antd';
+
+import './CarouselModal.scss';
+import { LeftOutlined, RightOutlined, CloseOutlined } from '@ant-design/icons';
+import useRefOrThis from '../../hooks/useRefOrThis';
+import { useGlobalEvent } from '../../hooks/useGlobalEvent';
+import useResponsiveCheck from '../../hooks/useResponsiveCheck';
+
+const CarouselModal = forwardRef(
```

Comment:
it's quite big, if we add more logic it would be better to split it

Reply:
I don't think it's big.. but I am happy to hear ideas how to split it, I think it wouldn't make so much difference and would be lots of prop/callback passing.